packages/flutter/lib/src/widgets/selection_area.dart

```diff
@@ -0,0 +1,211 @@
+
+  static SelectionRegistrant? of(BuildContext context) {
+    final _SelectionAreaState? registrant = context.findAncestorStateOfType<_SelectionAreaState>();
+    if (registrant == null || !registrant.widget.enabled) {
+      return null;
+    }
+    return registrant;
+  }
+}
+
+class _SelectionAreaState extends State<SelectionArea> implements SelectionRegistrant {
+  final GlobalKey<RawGestureDetectorState> _gestureDetectorKey = GlobalKey<RawGestureDetectorState>();
+  final Map<Type, GestureRecognizerFactory> _gestureRecognizers = <Type, GestureRecognizerFactory>{};
+  final Map<Selectable<Object>, Rect> _selectionCandidates = <Selectable<Object>, Rect>{};
+  final Set<Selectable<Object>> _selectionStates = <Selectable<Object>>{};
+
+  Offset? _selectionStart;
+  Offset? _selectionEnd;
+
+  @override
+  void update(Selectable<Object> selectable, Rect globalRect) {
+    _selectionCandidates[selectable] = globalRect;
+  }
+
+  @override
+  void remove(Selectable<Object> selectable) {
+    _selectionCandidates.remove(selectable);
+  }
+
+  void _handleDragDown(DragDownDetails details) {
+    _selectionStart = null;
+    _selectionEnd = null;
+  }
+
+  void _handleDragStart(DragStartDetails details) {
+    _selectionStart = details.globalPosition;
+  }
+
+  void _handleDragUpdate(DragUpdateDetails details) {
+    _selectionEnd = details.globalPosition;
+    _maybeUpdateSelection();
+  }
+
+  void _handleDragEnd(DragEndDetails details) {}
+
+  void _handleDragCancel() {
+    _selectionStart = null;
+    _selectionEnd = null;
+    _maybeUpdateSelection();
+  }
+
+  void _handleTapDown(TapDownDetails details) {
+    if (details.kind != PointerDeviceKind.mouse)
+      return;
+    _selectionStart = null;
+    _selectionEnd = null;
+    _maybeUpdateSelection();
+  }
+
+  // This logic needs to change based on text direction.
+  void _maybeUpdateSelection() {
+    final Offset? selectionStartT = _selectionStart;
+    final Offset? selectionEndT = _selectionEnd;
+    if (selectionEndT == null || selectionStartT == null) {
+      if (_selectionStates.isNotEmpty) {
+        for (final Selectable<Object> selectable in _selectionStates) {
+          selectable.clear();
+        }
+        _selectionStates.clear();
+      }
+      return;
+    }
+    final Offset selectionStart = selectionStartT;
+    final Offset selectionEnd = selectionEndT;
+    for (final Selectable<Object> data in _selectionCandidates.keys) {
+      final Rect? bounds = _selectionCandidates[data];
+      if (bounds == null) {
+        continue;
+      }
+      if (data.update(selectionStart, selectionEnd)) {
+        _selectionStates.add(data);
+      }
+    }
+  }
+
+  void _onKeyEvent(RawKeyEvent event) {
+    if (event.isControlPressed && event.isKeyPressed(LogicalKeyboardKey.keyC)) {
+      _onCopy();
+    }
+  }
+
+  void _onCopy() {
+    if (_selectionStates.isEmpty)
+      return;
+    // The order in which these should be concatenated is ???. For now this
```

Comment:
I actually came here to ask the same question. How do we determine the order the content of the "selected" widgets should appear in in the copy buffer?

Reply:
Yes, this one is a real quandary! Right now I am sorting by the topLeft offset of the render objects bounding rect. Something we can talk about today - but there is probably room for gradually adjusting this algorithm based on feedback

Reply:
I tried on html, it seems like it uses div order https://www.w3schools.com/code/tryit.asp?filename=GQ94272JBPA3
Maybe we can just use widget order and web developers may already be used to it?

Reply:
Definitely worth trying - though what is interesting in that example is how the selection boxes are calculated. Maybe this is actually more of a hit test special case

Reply:
Each widget (widget's renderer) could report their selection value and the rectangle within the overall selection rectangle that their selection value came from. The overall selection widget could assemble the contributions top to bottom and according to the text direction.